src/main/java/com/puppycrawl/tools/checkstyle/api/FileContents.java

```diff
@@ -154,16 +154,14 @@ public void reportBlockComment(int startLineNo, int startColNo,
                 endColNo);
 
         // save the comment
-        if (clangComments.containsKey(startLineNo)) {
-            final List<TextBlock> entries = clangComments.get(startLineNo);
-            entries.add(comment);
-        }
-        else {
-            final List<TextBlock> entries = new ArrayList<>();
-            entries.add(comment);
+        List<TextBlock> entries = clangComments.get(startLineNo);
+        if (entries == null) {
```

Comment:
This can be simplified with `computeIfAbsent`:

```
    clangComments.computeIfAbsent(startLineNo, key -> new ArrayList<>())
        .add(comment);
```

Reply:
Sonar saw it and reported it too. Done.